### ISSUE TITLE
Ordering/sorting table views

### ISSUE THREAD

**Issue by Ritmeester** [heartexlabs/label-studio#3586]
**Is your feature request related to a problem? Please describe.**
We have created a project in which we want to present several different dimensions from different data sources in a single table. The order of the rows in the table seems completely arbitrary at this moment. The order is not the same as it is in the input JSON file.

**Describe the solution you'd like**
The default behaviour that the order in which you upload the data is also the order in which the rows appear in the table. After a lot of head scratching, it seems table keys are now ordered **by the length of the key**, which is... an interesting design choice.

**Describe alternatives you've considered**
There are none.

**Additional context**
At this point, the Visual interface editor supposedly allows you to pick an input field, but the chosen input field is still not used in the preview of the UI. Instead, your dummy data is used instead.

Version 1.4.1post1

**Comment by makseq** [comment]
@Ritmeester sorry, it's very unclear which rows / order / keys / input field you mean. Could you please provide some screenshots with clarifications?

**Comment by Ritmeester** [comment]
<img width="538" alt="lstest" src="https://user-images.githubusercontent.com/113346155/215738624-ecef4607-bdf7-4d44-9cdb-f7ebe9ab1100.png">

Interface code:
```
<View>
  <Table name="table" value="$input"/>
  <TextArea name="text" toName="table"/>
</View>
```

Test input data:
```
{
    "id": 1,
    "input": {
        "aFirst": "test",
        "bSecond": "test",
        "cThird": "test"
    }
}
```

Rows: table rows
Keys: left column
Order: sorted order of the left column

Expected behaviour: maintain the key order as in the input JSON: 
`1. aFirst, 2. bSecond, 3. cThird`
Actual behaviour: keys are somehow sorted by length: 
`1. aFirst (shortest), 2. cThird (medium), 3. bSecond (longest).`
Required workaround: pad the names to force the order with invisible characters (denoted here with '!'): 
`1. aFirst, 2. bSecond, 3. cThird!!`

**Comment by makseq** [comment]
@Ritmeester Unfortunately the order of JSON fields is not preserved by JSON standards - https://stackoverflow.com/questions/7214293/is-the-order-of-elements-in-a-json-list-preserved.

You can try to reformat your input data and use multiple Text tags instead. Something like [this one](https://labelstud.io/playground/?config=%3CView%3E%3Cbr%3E%20%20%3CHeader%20value%3D%22Trick%20to%20build%20a%20table%22%2F%3E%3Cbr%3E%3Cbr%3E%20%20%3CView%20style%3D%22display%3A%20grid%3B%20%20grid-template-columns%3A%201fr%201fr%3B%20max-height%3A%20300px%3B%20width%3A%20400px%22%3E%3Cbr%3E%20%20%20%20%3CText%20name%3D%22key1%22%20placeholder%3D%22name%22%20value%3D%22%24key1%22%20%2F%3E%3Cbr%3E%20%20%20%20%3CText%20name%3D%22value1%22%20placeholder%3D%22name%22%20value%3D%22%24value1%22%20%2F%3E%3Cbr%3E%20%20%20%20%3Cbr%3E%20%20%20%20%3CText%20name%3D%22key2%22%20placeholder%3D%22name%22%20value%3D%22%24key2%22%20%2F%3E%3Cbr%3E%20%20%20%20%3CText%20name%3D%22value2%22%20placeholder%3D%22name%22%20value%3D%22%24value2%22%20%2F%3E%3Cbr%3E%20%20%20%20%3Cbr%3E%20%20%20%20%3CText%20name%3D%22key3%22%20placeholder%3D%22name%22%20value%3D%22%24key3%22%20%2F%3E%3Cbr%3E%20%20%20%20%3CText%20name%3D%22value3%22%20placeholder%3D%22name%22%20value%3D%22%24value3%22%20%2F%3E%3Cbr%3E%20%20%3C%2FView%3E%3Cbr%3E%3C%2FView%3E).

**Comment by Ritmeester** [comment]
That's unfortunate, but even so, it is a bit odd that there is no ordering/sorting available in the table view. People generally do not use tables for displaying their rows in a random order. Adding this would make a very valuable addition. The workaround presented above is just that, a workaround, not so much a solution imo.

**Comment by makseq** [comment]
Agree, now it sounds more like feature request like: order "Table items by keys". We will discuss it with our team, sound like easy thing to do. Ticket has been created: LSDV-4613

**Comment by makseq** [comment]
@Ritmeester please, try LS from the develop branch now, it should be fixed there and will be included in the release 1.7.2
